app/models/answer.rb

```diff
@@ -21,8 +21,35 @@ class Answer < ApplicationRecord
 
   validates :text, presence: true
 
-  # set default status to unanswered
+  # Set default status to unanswered
   after_initialize do
     self.status ||= :unanswered if new_record?
   end
+
+  ##
+  # When a user is in delegation mode and hits a checkbox next to a question, a
+  # request is sent to mark the question's answer as "predelegated", and the
+  # email of the delegated user is stored in the answer's text
+  #
+  def set_status_predelegated(email)
+    self.status = :predelegated
+    self.text = email
```

Comment:
so these won't actually commit to the database I believe. I think you need to `.save`. Because of this, if any errors occur you actually need to propagate them back up so that the controller and send an error json message. This function may actually be better in the controller since you're directly making changes to the database off of a network request.

Reply:
this should but be a put method in the controller I think, just because the input isn't controlled by us unlike set_status_delegated. `email` is something the user passes in, and it needs to be validated and potentially throw an error if the email passed in looks not like an email.